ql/termstructures/yield/ratehelpers.cpp

```diff
@@ -36,10 +37,118 @@
 #include <ql/time/calendars/unitedstates.hpp>
 #include <ql/time/imm.hpp>
 #include <ql/utilities/null_deleter.hpp>
-#include <ql/optional.hpp>
 #include <utility>
 
+#define APPLY_DATES(dates)                               \
```

Comment:
If I did not miss something, you are always using this macro like
```c++
const FuturesRateHelperDates dates = InitializeDates(...);
APPLY_DATES(dates);
```
In this case it will be simpler to make `InitializeDates` into private methods that directly set the fields. Then you don't need this macro and the `FuturesRateHelperDates` struct.

Reply:
If we use delegating constructors, the initialization code is no longer duplicated and can go in the constructors.  I'll push the changes.